pkg/daemon/update.go

```diff
@@ -397,7 +418,34 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 		}
 	}()
 
-	return dn.updateOSAndReboot(newConfig)
+	if err := dn.updateOS(newConfig); err != nil {
+		return err
+	}
+
+	for _, action := range actions {
+		// There is no harm in reconfirming that the node is drained
+		glog.Infof("Performing %v", action.Describe(dn))
+		if err := action.Execute(dn, newConfig); err != nil {
+			glog.Errorf("Applying machine config failed, node will reboot: %v", err)
+			return dn.finalizeAndReboot(newConfig)
+		}
+	}
+
+	// If we are not rebooting, we need to replicate the
+	// checkStateOnFirstRun() functionality to indicate that the node has
+	// been successfully updated
+	if err := dn.nodeWriter.SetDone(
+		dn.kubeClient.CoreV1().Nodes(),
+		dn.nodeLister,
+		dn.name,
+		newConfigName,
+	); err != nil {
+		glog.Errorf("Setting node's state to Done failed, node will reboot: %v", err)
+		return dn.drainAndReboot(newConfig)
+	}
```

Comment:
will play with this PR a little bit - not fully sure about the consequences of having this here, what happens when rebooting? 🤔 - Also, as safe as any changes can be, should we cordon/drain/uncordon anyway?

Reply:
I wouldn't be averse to always performing a cordon/drain/uncordon cycle, it would make the code a little simpler.

I didn't look closely enough at the reboot case to know why this block isn't needed there.

Reply:
Updated PR unconditionally drains and automatically uncordons if not rebooting.